pkg/service/customer.go

```diff
@@ -44,17 +57,42 @@ func (c *Customer) Start(ctx context.Context) error {
 	if err != nil {
 		return fmt.Errorf("falha ao iniciar o servidor GRPC: %w", err)
 	}
-	if err = c.grpc.Start(ctx); err != nil {
-		return fmt.Errorf("falha ao inicia o servidor GRPC: %w", err)
-	}
+	go func() {
+		if grpcErr := c.grpc.Start(ctx); grpcErr != nil && !errors.Is(grpcErr, grpc.ErrServerStopped) {
```

Comment:
Eu acho que você pode omitir o `grpcErr != nil`, já que o errors.Is deve aceitar um nil como parâmetro. Só inverta a ordem do errors.Is, para que a constante fique na primeira posição, de forma que se leia: errors.Is(esperado, obtido).